earth_enterprise/src/common/khTypes.h

```diff
@@ -62,6 +62,61 @@ typedef double                          float64;
 
 #ifdef __cplusplus
 
+// used to limit the number of strings in memory
+// thread-safe and removes when no longer needed
+#include <boost/flyweight.hpp>
+
+class AssetHandle
+{
+    boost::flyweights::flyweight<std::string> fw;
+public:
+    AssetHandle() = default;
+    AssetHandle(const AssetHandle& other) { fw = other.fw; }
+    AssetHandle(const std::string& ref) { fw = ref; }
+    AssetHandle(AssetHandle&& other) { fw = std::move(other.fw); }
+    AssetHandle(std::string&& ref) { fw = std::move(ref); }
+
+    AssetHandle& operator=(const AssetHandle& other)
+    {
+        fw = other.fw;
+        return *this;
+    }
+    AssetHandle& operator=(const std::string& ref)
+    {
+        fw = ref;
+        return *this;
+    }
+    AssetHandle& operator=(AssetHandle&& other)
+    {
+        fw = std::move(other.fw);
+        return *this;
+    }
+    AssetHandle& operator=(std::string&& ref)
+    {
+        fw = std::move(ref);
+        return *this;
+    }
+
+    friend std::ostream& operator<<(std::ostream& out,
+                                    const AssetHandle& other)
+    {
+        out << other.fw;
+        return out;
+    }
+
+    operator std::string() const { return std::string(fw); }
+    bool operator==(const AssetHandle& other) const { return fw == other.fw; }
+    bool operator!=(const AssetHandle& other) const { return fw != other.fw; }
+    bool operator<(const AssetHandle& other) const { return fw < other.fw; }
+    bool operator>(const AssetHandle& other) const { return fw > other.fw; }
+    bool operator==(const std::string& ref) const { return fw == ref; }
+    bool operator!=(const std::string& ref) const { return fw != ref; }
+    bool operator<(const std::string& ref) const { return fw < ref; }
+    bool operator>(const std::string& ref) const { return fw > ref; }
+
+    bool empty() const { return std::string(fw).size() == 0 ? true : false; }
```

Comment:
Minor: you don't need the `:?` operator here, you can just `return std::string(fw).size == 0;` or even `return std::string(fw).empty();`.

Thinking out loud here - for efficiency sake it might be faster to check `fw == boost::flyweights::flyweight<std::string>("");` since comparisons between flyweights are likely more efficient than creating a new string.